spacy_llm/tasks/spancat.py

```diff
@@ -0,0 +1,106 @@
+Below are some examples (only use these as a guide):
+{# whitespace #}
+{# whitespace #}
+{%- for example in examples -%}
+{# whitespace #}
+Text:
+'''
+{{ example.text }}
+'''
+{# whitespace #}
+{%- for label, substrings in example.entities.items() -%}
+{{ label }}: {{ ', '.join(substrings) }}
+{# whitespace #}
+{%- endfor -%}
+{# whitespace #}
+{# whitespace #}
+{%- endfor -%}
+{%- endif -%}
+{# whitespace #}
+Here is the text that needs labeling:
+{# whitespace #}
+Text:
+'''
+{{ text }}
+'''
+    """
+
+    def __init__(
+        self,
+        labels: str,
+        spans_key: str = "sc",
+        examples: Optional[Callable[[], Iterable[Any]]] = None,
+        normalizer: Optional[Callable[[str], str]] = None,
+        alignment_mode: Literal[
+            "strict", "contract", "expand"  # noqa: F821
+        ] = "contract",
+        case_sensitive_matching: bool = False,
+        single_match: bool = False,
+    ):
+        """Default NER task.
+
+        labels (str): Comma-separated list of labels to pass to the template.
+        spans_key (str): Key of the `Doc.spans` dict to save under.
+        examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
+            reads a file containing task examples for few-shot learning. If None is
+            passed, then zero-shot learning will be used.
+        normalizer (Optional[Callable[[str], str]]): optional normalizer function.
+        alignment_mode (str): "strict", "contract" or "expand".
+        case_sensitive: Whether to search without case sensitivity.
+        single_match (bool): If False, allow one substring to match multiple times in
+            the text. If True, returns the first hit.
+        """
+        super(SpanCatTask, self).__init__(
+            labels=labels,
+            examples=examples,
+            normalizer=normalizer,
+            alignment_mode=alignment_mode,
+            case_sensitive_matching=case_sensitive_matching,
+            single_match=single_match,
+        )
+        self._span_key = spans_key
+
+    def parse_responses(
+        self, docs: Iterable[Doc], responses: Iterable[str]
+    ) -> Iterable[Doc]:
+        for doc, prompt_response in zip(docs, responses):
+            spans = []
+            for label, phrases in self._format_response(prompt_response):
+                # For each phrase, find the substrings in the text
+                # and create a Span
+                offsets = find_substrings(
+                    doc.text,
+                    phrases,
+                    case_sensitive=self._case_sensitive_matching,
+                    single_match=self._single_match,
+                )
+                for start, end in offsets:
+                    span = doc.char_span(
+                        start, end, alignment_mode=self._alignment_mode, label=label
+                    )
+                    if span is not None:
+                        spans.append(span)
+            doc.spans[self._span_key] = spans
```

Comment:
I think everything except this final assignment line should be shared with the NER task?